{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Change this function to select a different paper set\n",
    "def read_metadata(path):\n",
    "    df = pd.read_csv(path)\n",
    "    df = df.astype({'abstract': 'str'})\n",
    "    df = df[df['abstract'] != 'Unknown']\n",
    "    df = df[df['abstract'] != 'nan']\n",
    "    #df = df[df['language'] == 'en']\n",
    "    #df = df[df['publish_time'].str[:4] == '2020']\n",
    "    df = df.drop_duplicates(['title','abstract'])\n",
    "    #df = df.drop_duplicates(['title'])\n",
    "    #df = df.drop_duplicates(['abstract'])\n",
    "    df = df.reset_index(drop=True)\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "#METADATA_PATH = 'data/metadata_new_new.csv'\n",
    "METADATA_PATH = 'data/metadata_processed.csv'\n",
    "\n",
    "df_meta = read_metadata(METADATA_PATH)\n",
    "abstracts = [a for a in df_meta['abstract']]\n",
    "titles = df_meta['title'].reset_index(drop=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "title_values = df_meta['title'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Replace duplicate title with: title v0, title v1, title v2, ...\n",
    "for title, n in title_values.iteritems():\n",
    "    if n >= 2:\n",
    "        i = 0\n",
    "        for idx, series in df_meta[df_meta['title'] == title].iterrows():\n",
    "            series['title'] = series['title'] + ' #' + repr(i)\n",
    "            df_meta.iloc[idx] = series\n",
    "            i += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Real-Time Reverse Transcription PCR Assay for Detection of Senecavirus A in Swine Vesicular Diagnostic Specimens                                         1\n",
       "Bat Airway Epithelial Cells: A Novel Tool for the Study of Zoonotic Viruses                                                                              1\n",
       "Spatial epidemiology of networked metapopulation: An overview                                                                                            1\n",
       "Isolation and molecular characterization of Sul/01/09 avian infectious bronchitis virus, indicates the emergence of a new genotype in the Middle East    1\n",
       "Study and interest of cellular load in respiratory samples for the optimization of molecular virological diagnosis in clinical practice                  1\n",
       "                                                                                                                                                        ..\n",
       "Cocaine: On-line analysis of an accumbens amine neural basis for psychomotor behavior                                                                    1\n",
       "Characterization of EV-2, a Virus Isolated from European Eels (Anguilla anguilla) with Stomatopapilloma                                                  1\n",
       "The SARS coronavirus nucleocapsid protein – Forms and functions                                                                                          1\n",
       "Semliki Forest virus-induced polykaryocyte formation is an ATP-dependent event                                                                           1\n",
       "Removal of Hepatitis C Virus-Infected Cells by a Zymogenized Bacterial Toxin                                                                             1\n",
       "Name: title, Length: 100, dtype: int64"
      ]
     },
     "execution_count": 149,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta['title'].value_counts()[:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>url</th>\n",
       "      <th>language</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>vho70jcx</td>\n",
       "      <td>f056da9c64fbf00a4645ae326e8a4339d015d155</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>SIANN: Strain Identification by Alignment to N...</td>\n",
       "      <td>10.1101/001727</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Next-generation sequencing is increasingly bei...</td>\n",
       "      <td>2014-01-10</td>\n",
       "      <td>Samuel Minot; Stephen D Turner; Krista L Ternu...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "      <td>https://doi.org/10.1101/001727</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>i9tbix2v</td>\n",
       "      <td>daf32e013d325a6feb80e83d15aabc64a48fae33</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Spatial epidemiology of networked metapopulati...</td>\n",
       "      <td>10.1101/003889</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>An emerging disease is one infectious epidemic...</td>\n",
       "      <td>2014-06-04</td>\n",
       "      <td>Lin WANG; Xiang Li</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "      <td>https://doi.org/10.1101/003889</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>62gfisc6</td>\n",
       "      <td>f33c6d94b0efaa198f8f3f20e644625fa3fe10d2</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Sequencing of the human IG light chain loci fr...</td>\n",
       "      <td>10.1101/006866</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Germline variation at immunoglobulin gene (IG)...</td>\n",
       "      <td>2014-07-03</td>\n",
       "      <td>Corey T Watson; Karyn Meltz Steinberg; Tina A ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "      <td>https://doi.org/10.1101/006866</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>058r9486</td>\n",
       "      <td>4da8a87e614373d56070ed272487451266dce919</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Bayesian mixture analysis for metagenomic comm...</td>\n",
       "      <td>10.1101/007476</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Deep sequencing of clinical samples is now an ...</td>\n",
       "      <td>2014-07-25</td>\n",
       "      <td>Sofia Morfopoulou; Vincent Plagnol</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "      <td>https://doi.org/10.1101/007476</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>wich35l7</td>\n",
       "      <td>eccef80cfbe078235df22398f195d5db462d8000</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Mapping a viral phylogeny onto outbreak trees ...</td>\n",
       "      <td>10.1101/010389</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biorxiv</td>\n",
       "      <td>Developing methods to reconstruct transmission...</td>\n",
       "      <td>2014-11-11</td>\n",
       "      <td>Stephen P Velsko; Jonathan E Allen</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "      <td>https://doi.org/10.1101/010389</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37502</th>\n",
       "      <td>45768</td>\n",
       "      <td>6b1y7yxg</td>\n",
       "      <td>f81692543d3e35858911cea48c298bfa23b20bc6</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Quality of life and psychological status in su...</td>\n",
       "      <td>10.1016/j.jpsychores.2005.08.020</td>\n",
       "      <td>PMC7094294</td>\n",
       "      <td>16650592.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Little is known about the long-term consequenc...</td>\n",
       "      <td>2006-05-31</td>\n",
       "      <td>Kwek, Seow-Khee; Chew, Wuen-Ming; Ong, Kian-Ch...</td>\n",
       "      <td>Journal of Psychosomatic Research</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/j.jpsychores.2005.08.020</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37503</th>\n",
       "      <td>45769</td>\n",
       "      <td>4360s2yu</td>\n",
       "      <td>289deae0b2050aa259a05ba84565a4df82fa099a</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Personal Protective Equipment: Protecting Heal...</td>\n",
       "      <td>10.1016/j.clinthera.2015.07.007</td>\n",
       "      <td>PMC4661082</td>\n",
       "      <td>26452427.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>The recent Ebola epidemic that devastated West...</td>\n",
       "      <td>2015-11-01</td>\n",
       "      <td>Fischer, William A.; Weber, David J.; Wohl, Da...</td>\n",
       "      <td>Clinical Therapeutics</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/j.clinthera.2015.07.007</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37504</th>\n",
       "      <td>45770</td>\n",
       "      <td>66jumbir</td>\n",
       "      <td>21a4369f83891bf6975dd916c0aa495d5df8709e</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Viruses and asthma</td>\n",
       "      <td>10.1016/j.bbagen.2011.01.012</td>\n",
       "      <td>PMC3130828</td>\n",
       "      <td>21291960.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Viral respiratory infection has long been know...</td>\n",
       "      <td>2011-11-30</td>\n",
       "      <td>Dulek, Daniel E.; Peebles, R. Stokes</td>\n",
       "      <td>Biochimica et Biophysica Acta (BBA) - General ...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/j.bbagen.2011.01.012</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37505</th>\n",
       "      <td>45771</td>\n",
       "      <td>3wk36h9p</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Why the WHO won't use the p-word</td>\n",
       "      <td>10.1016/s0262-4079(20)30474-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>There are no criteria for a pandemic, but covi...</td>\n",
       "      <td>2020-03-07</td>\n",
       "      <td>MacKenzie, Debora</td>\n",
       "      <td>New Scientist</td>\n",
       "      <td>NaN</td>\n",
       "      <td>#5716</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/s0262-4079(20)30474-7</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37506</th>\n",
       "      <td>45773</td>\n",
       "      <td>28vx9w58</td>\n",
       "      <td>3369a14e1d116943f48b3a33597796c9802de279; f523...</td>\n",
       "      <td>PMC</td>\n",
       "      <td>Searching for animal models and potential targ...</td>\n",
       "      <td>10.1016/j.onehlt.2017.03.001</td>\n",
       "      <td>PMC5454147</td>\n",
       "      <td>28616501.0</td>\n",
       "      <td>cc-by-nc-nd</td>\n",
       "      <td>Emerging and re-emerging pathogens represent a...</td>\n",
       "      <td>2017-03-03</td>\n",
       "      <td>Vergara-Alert, Júlia; Vidal, Enric; Bensaid, A...</td>\n",
       "      <td>One Health</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>noncomm_use_subset</td>\n",
       "      <td>https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37507 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  cord_uid  \\\n",
       "0               0  vho70jcx   \n",
       "1               1  i9tbix2v   \n",
       "2               2  62gfisc6   \n",
       "3               3  058r9486   \n",
       "4               4  wich35l7   \n",
       "...           ...       ...   \n",
       "37502       45768  6b1y7yxg   \n",
       "37503       45769  4360s2yu   \n",
       "37504       45770  66jumbir   \n",
       "37505       45771  3wk36h9p   \n",
       "37506       45773  28vx9w58   \n",
       "\n",
       "                                                     sha  source_x  \\\n",
       "0               f056da9c64fbf00a4645ae326e8a4339d015d155   biorxiv   \n",
       "1               daf32e013d325a6feb80e83d15aabc64a48fae33   biorxiv   \n",
       "2               f33c6d94b0efaa198f8f3f20e644625fa3fe10d2   biorxiv   \n",
       "3               4da8a87e614373d56070ed272487451266dce919   biorxiv   \n",
       "4               eccef80cfbe078235df22398f195d5db462d8000   biorxiv   \n",
       "...                                                  ...       ...   \n",
       "37502           f81692543d3e35858911cea48c298bfa23b20bc6  Elsevier   \n",
       "37503           289deae0b2050aa259a05ba84565a4df82fa099a  Elsevier   \n",
       "37504           21a4369f83891bf6975dd916c0aa495d5df8709e  Elsevier   \n",
       "37505                                                NaN  Elsevier   \n",
       "37506  3369a14e1d116943f48b3a33597796c9802de279; f523...       PMC   \n",
       "\n",
       "                                                   title  \\\n",
       "0      SIANN: Strain Identification by Alignment to N...   \n",
       "1      Spatial epidemiology of networked metapopulati...   \n",
       "2      Sequencing of the human IG light chain loci fr...   \n",
       "3      Bayesian mixture analysis for metagenomic comm...   \n",
       "4      Mapping a viral phylogeny onto outbreak trees ...   \n",
       "...                                                  ...   \n",
       "37502  Quality of life and psychological status in su...   \n",
       "37503  Personal Protective Equipment: Protecting Heal...   \n",
       "37504                                 Viruses and asthma   \n",
       "37505                   Why the WHO won't use the p-word   \n",
       "37506  Searching for animal models and potential targ...   \n",
       "\n",
       "                                    doi       pmcid   pubmed_id      license  \\\n",
       "0                        10.1101/001727         NaN         NaN      biorxiv   \n",
       "1                        10.1101/003889         NaN         NaN      biorxiv   \n",
       "2                        10.1101/006866         NaN         NaN      biorxiv   \n",
       "3                        10.1101/007476         NaN         NaN      biorxiv   \n",
       "4                        10.1101/010389         NaN         NaN      biorxiv   \n",
       "...                                 ...         ...         ...          ...   \n",
       "37502  10.1016/j.jpsychores.2005.08.020  PMC7094294  16650592.0    els-covid   \n",
       "37503   10.1016/j.clinthera.2015.07.007  PMC4661082  26452427.0    els-covid   \n",
       "37504      10.1016/j.bbagen.2011.01.012  PMC3130828  21291960.0    els-covid   \n",
       "37505     10.1016/s0262-4079(20)30474-7         NaN         NaN    els-covid   \n",
       "37506      10.1016/j.onehlt.2017.03.001  PMC5454147  28616501.0  cc-by-nc-nd   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "0      Next-generation sequencing is increasingly bei...   2014-01-10   \n",
       "1      An emerging disease is one infectious epidemic...   2014-06-04   \n",
       "2      Germline variation at immunoglobulin gene (IG)...   2014-07-03   \n",
       "3      Deep sequencing of clinical samples is now an ...   2014-07-25   \n",
       "4      Developing methods to reconstruct transmission...   2014-11-11   \n",
       "...                                                  ...          ...   \n",
       "37502  Little is known about the long-term consequenc...   2006-05-31   \n",
       "37503  The recent Ebola epidemic that devastated West...   2015-11-01   \n",
       "37504  Viral respiratory infection has long been know...   2011-11-30   \n",
       "37505  There are no criteria for a pandemic, but covi...   2020-03-07   \n",
       "37506  Emerging and re-emerging pathogens represent a...   2017-03-03   \n",
       "\n",
       "                                                 authors  \\\n",
       "0      Samuel Minot; Stephen D Turner; Krista L Ternu...   \n",
       "1                                     Lin WANG; Xiang Li   \n",
       "2      Corey T Watson; Karyn Meltz Steinberg; Tina A ...   \n",
       "3                     Sofia Morfopoulou; Vincent Plagnol   \n",
       "4                     Stephen P Velsko; Jonathan E Allen   \n",
       "...                                                  ...   \n",
       "37502  Kwek, Seow-Khee; Chew, Wuen-Ming; Ong, Kian-Ch...   \n",
       "37503  Fischer, William A.; Weber, David J.; Wohl, Da...   \n",
       "37504               Dulek, Daniel E.; Peebles, R. Stokes   \n",
       "37505                                  MacKenzie, Debora   \n",
       "37506  Vergara-Alert, Júlia; Vidal, Enric; Bensaid, A...   \n",
       "\n",
       "                                                 journal  \\\n",
       "0                                                    NaN   \n",
       "1                                                    NaN   \n",
       "2                                                    NaN   \n",
       "3                                                    NaN   \n",
       "4                                                    NaN   \n",
       "...                                                  ...   \n",
       "37502                  Journal of Psychosomatic Research   \n",
       "37503                              Clinical Therapeutics   \n",
       "37504  Biochimica et Biophysica Acta (BBA) - General ...   \n",
       "37505                                      New Scientist   \n",
       "37506                                         One Health   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence  has_full_text  \\\n",
       "0                              NaN            NaN           True   \n",
       "1                              NaN            NaN           True   \n",
       "2                              NaN            NaN           True   \n",
       "3                              NaN            NaN           True   \n",
       "4                              NaN            NaN           True   \n",
       "...                            ...            ...            ...   \n",
       "37502                          NaN            NaN           True   \n",
       "37503                          NaN            NaN           True   \n",
       "37504                          NaN            NaN           True   \n",
       "37505                          NaN          #5716          False   \n",
       "37506                          NaN            NaN           True   \n",
       "\n",
       "           full_text_file                                                url  \\\n",
       "0         biorxiv_medrxiv                     https://doi.org/10.1101/001727   \n",
       "1         biorxiv_medrxiv                     https://doi.org/10.1101/003889   \n",
       "2         biorxiv_medrxiv                     https://doi.org/10.1101/006866   \n",
       "3         biorxiv_medrxiv                     https://doi.org/10.1101/007476   \n",
       "4         biorxiv_medrxiv                     https://doi.org/10.1101/010389   \n",
       "...                   ...                                                ...   \n",
       "37502      custom_license   https://doi.org/10.1016/j.jpsychores.2005.08.020   \n",
       "37503      custom_license    https://doi.org/10.1016/j.clinthera.2015.07.007   \n",
       "37504      custom_license       https://doi.org/10.1016/j.bbagen.2011.01.012   \n",
       "37505      custom_license      https://doi.org/10.1016/s0262-4079(20)30474-7   \n",
       "37506  noncomm_use_subset  https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5...   \n",
       "\n",
       "      language  \n",
       "0           en  \n",
       "1           en  \n",
       "2           en  \n",
       "3           en  \n",
       "4           en  \n",
       "...        ...  \n",
       "37502       en  \n",
       "37503       en  \n",
       "37504       en  \n",
       "37505       en  \n",
       "37506       en  \n",
       "\n",
       "[37507 rows x 19 columns]"
      ]
     },
     "execution_count": 150,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>url</th>\n",
       "      <th>language</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>25466</th>\n",
       "      <td>28697</td>\n",
       "      <td>z43ifmr3</td>\n",
       "      <td>43de7c4be57c2998c8e915ac6e943c176da10d20</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Emerging and Reemerging Virus Diseases of Vert...</td>\n",
       "      <td>10.1016/b978-012374410-4.00383-6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>In the last two decades, a large number of new...</td>\n",
       "      <td>2008-12-31</td>\n",
       "      <td>Mahy, B.W.J.</td>\n",
       "      <td>Encyclopedia of Virology</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/b978-012374410-4.00383-6</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25850</th>\n",
       "      <td>29190</td>\n",
       "      <td>25rblhwb</td>\n",
       "      <td>94477fab8df0d830a2a4dbada5eb3d62217687fd</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Emerging and Reemerging Virus Diseases of Vert...</td>\n",
       "      <td>10.1016/b978-0-12-801238-3.02564-2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>In the last two decades, a large number of new...</td>\n",
       "      <td>2014-12-31</td>\n",
       "      <td>Mahy, B.W.J</td>\n",
       "      <td>Reference Module in Biomedical Sciences</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/b978-0-12-801238-3.025...</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  cord_uid                                       sha  \\\n",
       "25466       28697  z43ifmr3  43de7c4be57c2998c8e915ac6e943c176da10d20   \n",
       "25850       29190  25rblhwb  94477fab8df0d830a2a4dbada5eb3d62217687fd   \n",
       "\n",
       "       source_x                                              title  \\\n",
       "25466  Elsevier  Emerging and Reemerging Virus Diseases of Vert...   \n",
       "25850  Elsevier  Emerging and Reemerging Virus Diseases of Vert...   \n",
       "\n",
       "                                      doi pmcid  pubmed_id    license  \\\n",
       "25466    10.1016/b978-012374410-4.00383-6   NaN        NaN  els-covid   \n",
       "25850  10.1016/b978-0-12-801238-3.02564-2   NaN        NaN  els-covid   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "25466  In the last two decades, a large number of new...   2008-12-31   \n",
       "25850  In the last two decades, a large number of new...   2014-12-31   \n",
       "\n",
       "            authors                                  journal  \\\n",
       "25466  Mahy, B.W.J.                 Encyclopedia of Virology   \n",
       "25850   Mahy, B.W.J  Reference Module in Biomedical Sciences   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence  has_full_text  \\\n",
       "25466                          NaN            NaN           True   \n",
       "25850                          NaN            NaN           True   \n",
       "\n",
       "       full_text_file                                                url  \\\n",
       "25466  custom_license   https://doi.org/10.1016/b978-012374410-4.00383-6   \n",
       "25850  custom_license  https://doi.org/10.1016/b978-0-12-801238-3.025...   \n",
       "\n",
       "      language  \n",
       "25466       en  \n",
       "25850       en  "
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['abstract'] == \"In the last two decades, a large number of new viruses have been discovered, many of which are pathogenic in humans or other vertebrates. Among the more important causes of virus emergence have been changes in human behavior, population, and increases in travel to distant countries. In addition, the application of new molecular technologies has led to the recognition of many viruses that hitherto went undetected. Many of the new, emerging viruses have an RNA genome, and many are zoonoses. The spread of human immunodeficiency virus, causing acquired immune deficiency syndrome, and the use of immunosuppressive drugs following transplant surgery, have increased the numbers of people in the population that are highly susceptible to emerging virus infections. The threat of a new pandemic of influenza virus in the human population stresses the need for development of better methods for detection, surveillance, and control of emerging virus diseases.\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Unnamed: 0</th>\n",
       "      <th>cord_uid</th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>url</th>\n",
       "      <th>language</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27663</th>\n",
       "      <td>32504</td>\n",
       "      <td>r94tu8fi</td>\n",
       "      <td>f085aa777d230139f3ae371f7d28ec3070a6c254</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>The Molecular Biology of Coronaviruses #0</td>\n",
       "      <td>10.1016/s0065-3527(06)66005-3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Coronaviruses are large, enveloped RNA viruses...</td>\n",
       "      <td>2006-12-31</td>\n",
       "      <td>Masters, Paul S.</td>\n",
       "      <td>Advances in Virus Research</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>https://doi.org/10.1016/s0065-3527(06)66005-3</td>\n",
       "      <td>en</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Unnamed: 0  cord_uid                                       sha  \\\n",
       "27663       32504  r94tu8fi  f085aa777d230139f3ae371f7d28ec3070a6c254   \n",
       "\n",
       "       source_x                                      title  \\\n",
       "27663  Elsevier  The Molecular Biology of Coronaviruses #0   \n",
       "\n",
       "                                 doi pmcid  pubmed_id    license  \\\n",
       "27663  10.1016/s0065-3527(06)66005-3   NaN        NaN  els-covid   \n",
       "\n",
       "                                                abstract publish_time  \\\n",
       "27663  Coronaviruses are large, enveloped RNA viruses...   2006-12-31   \n",
       "\n",
       "                authors                     journal  \\\n",
       "27663  Masters, Paul S.  Advances in Virus Research   \n",
       "\n",
       "       Microsoft Academic Paper ID WHO #Covidence  has_full_text  \\\n",
       "27663                          NaN            NaN           True   \n",
       "\n",
       "       full_text_file                                            url language  \n",
       "27663  custom_license  https://doi.org/10.1016/s0065-3527(06)66005-3       en  "
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta[df_meta['title'] == 'The Molecular Biology of Coronaviruses #0']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Emerging and Reemerging Virus Diseases of Vertebrates'"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.iloc[25466]['title']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Emerging and Reemerging Virus Diseases of Vertebrates☆'"
      ]
     },
     "execution_count": 154,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.iloc[25850]['title']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Coronaviruses are large, enveloped RNA viruses of both medical and veterinary importance. Interest in this viral family has intensified in the past few years as a result of the identification of a newly emerged coronavirus as the causative agent of severe acute respiratory syndrome (SARS). At the molecular level, coronaviruses employ a variety of unusual strategies to accomplish a complex program of gene expression. Coronavirus replication entails ribosome frameshifting during genome translation, the synthesis of both genomic and multiple subgenomic RNA species, and the assembly of progeny virions by a pathway that is unique among enveloped RNA viruses. Progress in the investigation of these processes has been enhanced by the development of reverse genetic systems, an advance that was heretofore obstructed by the enormous size of the coronavirus genome. This review summarizes both classical and contemporary discoveries in the study of the molecular biology of these infectious agents, with particular emphasis on the nature and recognition of viral receptors, viral RNA synthesis, and the molecular interactions governing virion assembly.'"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.iloc[27663]['abstract']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Publisher Summary This chapter discusses the manipulation of clones of coronavirus and of complementary DNAs (cDNAs) of defective-interfering (DI) RNAs to study coronavirus RNA replication, transcription, recombination, processing and transport of proteins, virion assembly, identification of cell receptors for coronaviruses, and processing of the polymerase. The nature of the coronavirus genome is nonsegmented, single-stranded, and positive-sense RNA. Its size ranges from 27 to 32 kb, which is significantly larger when compared with other RNA viruses. The gene encoding the large surface glycoprotein is up to 4.4 kb, encoding an imposing trimeric, highly glycosylated protein. This soars some 20 nm above the virion envelope, giving the virus the appearance-with a little imagination-of a crown or coronet. Coronavirus research has contributed to the understanding of many aspects of molecular biology in general, such as the mechanism of RNA synthesis, translational control, and protein transport and processing. It remains a treasure capable of generating unexpected insights.'"
      ]
     },
     "execution_count": 156,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.iloc[27687]['abstract']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"Publisher Summary Coronaviruses have recently emerged as an important group of animal and human pathogens that share a distinctive replicative cycle. Some of the unique characteristics in the replication of coronaviruses include generation of a 3' coterminal-nested set of five or six subgenomic mRNAs, each of which appears to direct the synthesis of one protein. Two virus-specific RNA polymerase activities have been identified. Many of the distinctive features of coronavirus infection and coronavirus-induced diseases may result from the properties of the two coronavirus glycoproteins. The intracellular budding site, which may be important in the establishment and maintenance of persistent infections, appears to be due to the restricted intracytoplasmic migration of the E1 glycoprotein, which acts as a matrix-like transmembrane glycoprotein. E1 also exhibits distinctive behavior by self-aggregating on heating at 100°C in sodium dodecyl sulfate (SDS) and by its interaction with RNA in the viral nucleocapsid. The E1 of mouse hepatitis virus (MHV) is an O-linked glycoprotein, unlike most other viral glycoproteins. Thus, the coronavirus system may be a useful model for the study of synthesis, glycosylation, and transport of O-linked cellular glycoproteins.\""
      ]
     },
     "execution_count": 157,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta.iloc[27698]['abstract']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": [
    "#save the new dataframe\n",
    "new_metadata_path = 'data/metadata_sharp.csv'\n",
    "if os.path.isfile(new_metadata_path):\n",
    "    sys.exit('ERROR, a new dataset is already saved')\n",
    "else:\n",
    "    df_meta.to_csv(new_metadata_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
